src/main/java/edu/cornell/kfs/module/purap/document/service/impl/CuB2BShoppingServiceImpl.java

```diff
@@ -183,15 +190,48 @@ public List createRequisitionsFromCxml(B2BShoppingCart message, Person user) thr
 
             req.fixItemReferences();
 
-            // save requisition to database
-            purapService.saveDocumentNoValidation(req);
+            if (checkRequisitionAccountsAreUnique(req)) {
+                // save requisition to database
+                purapService.saveDocumentNoValidation(req);
 
-            // add requisition to List
-            requisitions.add(req);
+                // add requisition to List
+                requisitions.add(req);
+            }
+        }
+
+        if (Collections.isEmpty(requisitions)) {
+            throw new B2BShoppingException("Invalid e-shop Request. No requisitions were created.");
         }
+
         return requisitions;
     }
 
+    public boolean checkRequisitionAccountsAreUnique(RequisitionDocument requisitionDocument) {
+
+        List<RequisitionItem> requisitionItems = requisitionDocument.getItems();
+        boolean requisitionAccountsUnique = true;
+        for (RequisitionItem requisitionItem : requisitionItems) {
+
+            PurchasingAccountsPayableUniqueAccountingStringsValidation uniqueRequisitionAccountValidator =
+                    SpringContext.getBean(PurchasingAccountsPayableUniqueAccountingStringsValidation.class);
+            uniqueRequisitionAccountValidator.setItemForValidation(requisitionItem);
+
+            if (!uniqueRequisitionAccountValidator.validate(null)) {
```

Comment:
instead of using a NULL here, can you create a real event object?  I am just thinking about avoiding future NPEs if kualico changes their code in PurchasingAccountsPayableUniqueAccountingStringsValidation